### PR TITLE
Use ComparisonMode constants instead of magic numbers

### PR DESCRIPTION
**Pull request by mdickinson**
Minor cleanup PR: replace uses of magic numbers for comparison modes with the corresponding `ComparisonMode` elements. (The numbers should eventually become just an implementation detail; we  want to encourage all traits users to use the `ComparisonMode` constants instead.)

### CODE REVIEW

**Comment by mdickinson** [comment]
Note: this PR will conflict with #1165; whichever one gets merged second (probably this one) will need to deal with the merge conflicts.

**Comment by mdickinson** [comment]
Hmm. I somehow lost some of the changes when resolving conflicts. Will fix.

**Comment by mdickinson** [comment]
Ah no; I see - there were new cases introduced in #1208.

**Comment by mdickinson** [comment]
Updated after a master merge. Though now that #1165 is in, I'm wondering whether some of those comparison_mode=ComparisonMode.identity annotations can be removed. @kitchoi ?

File: traits/tests/test_observe.py [hunk 1]
```diff
@@ -417,12 +422,12 @@ class Potato(HasTraits):
 
 class PotatoBag(HasTraits):
 
-    potatos = List(Instance(Potato), comparison_mode=1)
+    potatos = List(Instance(Potato), comparison_mode=ComparisonMode.identity)
```
Comment (mdickinson) [inline traits/tests/test_observe.py:425]: I _really_ want to change this to `potatoes`, but that's out of scope for this PR.

Comment (kitchoi) [reply traits/tests/test_observe.py:425]: Oops. 😂

**Comment by kitchoi** [comment]
Race conditions from a number of PRs. Yes I agree the many `ComparisonMode.identity` aka `1`, can be removed.

**Comment by mdickinson** [comment]
@kitchoi Okay to merge this and look at removing redundant `comparison_mode=...` metadata later?

**Pull request merged**